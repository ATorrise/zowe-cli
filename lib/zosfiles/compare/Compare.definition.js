"use strict";
/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompareDefinition = void 0;
const Dataset_definition_1 = require("./ds/Dataset.definition");
const LocalfileDataset_definition_1 = require("./lf-ds/LocalfileDataset.definition");
const UssFile_definition_1 = require("./uss/UssFile.definition");
const LocalfileUss_definition_1 = require("./lf-uss/LocalfileUss.definition");
const Spooldd_definition_1 = require("./sdd/Spooldd.definition");
const LocalfileSpooldd_definition_1 = require("./lf-sdd/LocalfileSpooldd.definition");
// Does not use the import in anticipation of some internationalization work to be done later.
const strings = require("../-strings-/en").default.COMPARE;
/**
 * Compare group definition containing its description and children
 * @type {ICommandDefinition}
 */
exports.CompareDefinition = {
    name: "compare",
    aliases: ["cmp"],
    type: "group",
    summary: strings.SUMMARY,
    description: strings.DESCRIPTION,
    children: [
        Dataset_definition_1.DatasetDefinition,
        LocalfileDataset_definition_1.LocalfileDatasetDefinition,
        UssFile_definition_1.UssFileDefinition,
        LocalfileUss_definition_1.LocalfileUssFileDefinition,
        Spooldd_definition_1.SpoolddDefinition,
        LocalfileSpooldd_definition_1.LocalfileSpoolddDefinition
    ],
};
//# sourceMappingURL=Compare.definition.js.map