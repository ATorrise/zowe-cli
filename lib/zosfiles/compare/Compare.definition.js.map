{"version":3,"file":"Compare.definition.js","sourceRoot":"","sources":["../../../src/zosfiles/compare/Compare.definition.ts"],"names":[],"mappings":";AAAA;;;;;;;;;EASE;;;AAGF,gEAA0D;AAC1D,qFAAiF;AACjF,iEAA2D;AAC3D,8EAA8E;AAC9E,iEAA6D;AAC7D,sFAAkF;AAIlF,8FAA8F;AAC9F,MAAM,OAAO,GAAI,OAAO,CAAC,iBAAiB,CAAC,CAAC,OAA8B,CAAC,OAAO,CAAC;AAEnF;;;GAGG;AACU,QAAA,iBAAiB,GAAuB;IACjD,IAAI,EAAE,SAAS;IACf,OAAO,EAAE,CAAC,KAAK,CAAC;IAChB,IAAI,EAAE,OAAO;IACb,OAAO,EAAE,OAAO,CAAC,OAAO;IACxB,WAAW,EAAE,OAAO,CAAC,WAAW;IAChC,QAAQ,EAAE;QACN,sCAAiB;QACjB,wDAA0B;QAC1B,sCAAiB;QACjB,oDAA0B;QAC1B,sCAAiB;QACjB,wDAA0B;KAC7B;CACJ,CAAC"}