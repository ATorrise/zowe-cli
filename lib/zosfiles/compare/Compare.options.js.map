{"version":3,"file":"Compare.options.js","sourceRoot":"","sources":["../../../src/zosfiles/compare/Compare.options.ts"],"names":[],"mappings":";AAAA;;;;;;;;;EASE;;;AAMF,8FAA8F;AAC9F,MAAM,OAAO,GAAI,OAAO,CAAC,iBAAiB,CAAC,CAAC,OAA8B,CAAC,OAAO,CAAC,OAAO,CAAC;AAE3F;;GAEG;AACU,QAAA,cAAc,GAAgD;IAEvE;;;OAGG;IACH,MAAM,EAAE;QACJ,IAAI,EAAE,QAAQ;QACd,OAAO,EAAE,CAAC,GAAG,CAAC;QACd,WAAW,EAAE,OAAO,CAAC,MAAM;QAC3B,IAAI,EAAE,SAAS;KAClB;IAED;;;OAGG;IACH,OAAO,EAAE;QACL,IAAI,EAAE,SAAS;QACf,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,OAAO;QAC5B,IAAI,EAAE,SAAS;KAClB;IAED;;;OAGG;IACH,QAAQ,EAAE;QACN,IAAI,EAAE,UAAU;QAChB,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,QAAQ;QAC7B,IAAI,EAAE,QAAQ;KACjB;IAED;;;OAGG;IACH,SAAS,EAAE;QACP,IAAI,EAAE,WAAW;QACjB,OAAO,EAAE,CAAC,KAAK,CAAC;QAChB,WAAW,EAAE,OAAO,CAAC,SAAS;QAC9B,IAAI,EAAE,QAAQ;KACjB;IAED;;;OAGG;IACH,MAAM,EAAE;QACJ,IAAI,EAAE,QAAQ;QACd,OAAO,EAAE,CAAC,GAAG,CAAC;QACd,WAAW,EAAE,OAAO,CAAC,MAAM;QAC3B,IAAI,EAAE,SAAS;QACf,aAAa,EAAE,CAAC,QAAQ,CAAC;KAC5B;IAED;;;OAGG;IACH,OAAO,EAAE;QACL,IAAI,EAAE,SAAS;QACf,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,OAAO;QAC5B,IAAI,EAAE,SAAS;QACf,aAAa,EAAE,CAAC,SAAS,CAAC;KAC7B;IAED;;;OAGG;IAEH,MAAM,EAAE;QACJ,IAAI,EAAE,eAAe;QACrB,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,MAAM;QAC3B,IAAI,EAAE,QAAQ;KACjB;IAED;;;OAGG;IAEH,OAAO,EAAE;QACL,IAAI,EAAE,gBAAgB;QACtB,OAAO,EAAE,CAAC,KAAK,CAAC;QAChB,WAAW,EAAE,OAAO,CAAC,OAAO;QAC5B,IAAI,EAAE,QAAQ;KACjB;IAED;;;OAGG;IAEH,MAAM,EAAE;QACJ,IAAI,EAAE,QAAQ;QACd,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,MAAM;QAC3B,YAAY,EAAE,IAAI;QAClB,IAAI,EAAE,SAAS;KAClB;IAED;;;OAGG;IAEH,YAAY,EAAE;QACV,IAAI,EAAE,eAAe;QACrB,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,YAAY;QACjC,IAAI,EAAE,QAAQ;KACjB;IAED;;OAEG;IACH,WAAW,EAAE;QACT,IAAI,EAAE,cAAc;QACpB,OAAO,EAAE,CAAC,IAAI,CAAC;QACf,WAAW,EAAE,OAAO,CAAC,WAAW;QAChC,IAAI,EAAE,SAAS;KAClB;CACJ,CAAC"}