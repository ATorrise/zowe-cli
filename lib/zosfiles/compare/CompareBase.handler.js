"use strict";
/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompareBaseHandler = void 0;
const imperative_1 = require("@zowe/imperative");
const ZosFilesBase_handler_1 = require("../ZosFilesBase.handler");
const CompareBaseHelper_1 = require("./CompareBaseHelper");
/**
 * This class is used by the various zosfiles-compare handlers as the base class for their implementation.
 * All handlers within zosfiles-compare should extend this class.
 *
 * This class should not be used outside of the zosfiles-compare package.
 *
 * @private
 */
class CompareBaseHandler extends ZosFilesBase_handler_1.ZosFilesBaseHandler {
    /**
     * This will grab the zosmf profile and create a session before calling the subclass
     * {@link ZosFilesBaseHandler#processWithSession} method.
     *
     * @param {IHandlerParameters} commandParameters Command parameters sent by imperative.
     *
     * @returns {Promise<void>}
     */
    processWithSession(commandParameters, session) {
        return __awaiter(this, void 0, void 0, function* () {
            const helper = new CompareBaseHelper_1.CompareBaseHelper(commandParameters);
            helper.task = {
                percentComplete: 0,
                statusMessage: `Retrieving content for the first file/dataset`,
                stageName: imperative_1.TaskStage.IN_PROGRESS
            };
            commandParameters.response.progress.startBar({ task: helper.task });
            const fileContent1 = yield this.getFile1(session, commandParameters.arguments, helper);
            commandParameters.response.progress.endBar();
            commandParameters.response.progress.startBar({ task: helper.task });
            helper.task.statusMessage = `Retrieving content for the second file/dataset`;
            const fileContent2 = yield this.getFile2(session, commandParameters.arguments, helper);
            return helper.getResponse(helper.prepareContent(fileContent1), helper.prepareContent(fileContent2));
        });
    }
}
exports.CompareBaseHandler = CompareBaseHandler;
//# sourceMappingURL=CompareBase.handler.js.map