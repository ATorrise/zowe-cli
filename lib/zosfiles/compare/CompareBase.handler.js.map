{"version":3,"file":"CompareBase.handler.js","sourceRoot":"","sources":["../../../src/zosfiles/compare/CompareBase.handler.ts"],"names":[],"mappings":";AAAA;;;;;;;;;EASE;;;;;;;;;;;;AAEF,iDAAqG;AAErG,kEAA8D;AAC9D,2DAAwD;AAExD;;;;;;;GAOG;AACH,MAAsB,kBAAmB,SAAQ,0CAAmB;IAChE;;;;;;;OAOG;IACU,kBAAkB,CAAC,iBAAqC,EAAE,OAAwB;;YAC3F,MAAM,MAAM,GAAG,IAAI,qCAAiB,CAAC,iBAAiB,CAAC,CAAC;YACxD,MAAM,CAAC,IAAI,GAAG;gBACV,eAAe,EAAE,CAAC;gBAClB,aAAa,EAAE,+CAA+C;gBAC9D,SAAS,EAAE,sBAAS,CAAC,WAAW;aACnC,CAAC;YAEF,iBAAiB,CAAC,QAAQ,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE,IAAI,EAAE,MAAM,CAAC,IAAI,EAAE,CAAC,CAAC;YAEpE,MAAM,YAAY,GAAG,MAAM,IAAI,CAAC,QAAQ,CAAC,OAAO,EAAE,iBAAiB,CAAC,SAAS,EAAE,MAAM,CAAC,CAAC;YACvF,iBAAiB,CAAC,QAAQ,CAAC,QAAQ,CAAC,MAAM,EAAE,CAAC;YAC7C,iBAAiB,CAAC,QAAQ,CAAC,QAAQ,CAAC,QAAQ,CAAC,EAAE,IAAI,EAAE,MAAM,CAAC,IAAI,EAAE,CAAC,CAAC;YAEpE,MAAM,CAAC,IAAI,CAAC,aAAa,GAAG,gDAAgD,CAAC;YAC7E,MAAM,YAAY,GAAG,MAAM,IAAI,CAAC,QAAQ,CAAC,OAAO,EAAE,iBAAiB,CAAC,SAAS,EAAE,MAAM,CAAC,CAAC;YAEvF,OAAO,MAAM,CAAC,WAAW,CAAC,MAAM,CAAC,cAAc,CAAC,YAAY,CAAC,EAAE,MAAM,CAAC,cAAc,CAAC,YAAY,CAAC,CAAC,CAAC;QACxG,CAAC;KAAA;CAyBJ;AApDD,gDAoDC"}