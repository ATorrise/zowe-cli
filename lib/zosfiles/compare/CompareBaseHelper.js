"use strict";
/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CompareBaseHelper = void 0;
const path = require("path");
const fs = require("fs");
const imperative_1 = require("@zowe/imperative");
/**
 * Base helper for compare command to handle common operations through the comparison
 * @export
 */
class CompareBaseHelper {
    /**
     * Creates an instance of CompareBaseHelper
     * @param {IHandlerParameters} commandParameters Command parameters sent to the handler
     * @memberof CompareBaseHelper
     */
    constructor(commandParameters) {
        var _a;
        /**
         * file compare option for file1
         * @public
         * @memberof CompareBaseHelper
         */
        this.file1Options = {};
        /**
         * file compare option for file2
         * @public
         * @memberof CompareBaseHelper
         */
        this.file2Options = {};
        this.file1Options.binary = commandParameters.arguments.binary;
        this.file2Options.binary = commandParameters.arguments.binary2;
        this.file1Options.encoding = commandParameters.arguments.encoding;
        this.file2Options.encoding = commandParameters.arguments.encoding2;
        this.file1Options.record = commandParameters.arguments.record;
        this.file2Options.record = commandParameters.arguments.record2;
        this.file1Options.volumeSerial = commandParameters.arguments.volumeSerial;
        this.file2Options.volumeSerial = commandParameters.arguments.volumeSerial2;
        if (((_a = this.file2Options) === null || _a === void 0 ? void 0 : _a.binary) == null) {
            this.file2Options.binary = this.file1Options.binary;
        }
        if (this.file2Options.encoding == null) {
            this.file2Options.encoding = this.file1Options.encoding;
        }
        if (this.file2Options.record == null) {
            this.file2Options.record = this.file1Options.record;
        }
        this.seqnum = commandParameters.arguments.seqnum;
        this.browserView = commandParameters.arguments.browserView;
        this.contextLines = commandParameters.arguments.contextLines;
        this.responseTimeout = commandParameters.arguments.responseTimeout;
        this.file1Options.responseTimeout = this.responseTimeout;
        this.file2Options.responseTimeout = this.responseTimeout;
        // Clear undefined properties
        this.file1Options = JSON.parse(JSON.stringify(this.file1Options));
        this.file2Options = JSON.parse(JSON.stringify(this.file2Options));
    }
    /**
     * Parse the spool description and split them into individual properties
     * @param spoolDescription Colon-separated (:) spool descriptor
     * @returns Object containing the name, job id, and spool id
     */
    prepareSpoolDescriptor(spoolDescription) {
        const descriptionSeparator = ":";
        const spoolDescArr = spoolDescription.split(descriptionSeparator);
        const jobName = spoolDescArr[0];
        const jobId = spoolDescArr[1];
        const spoolId = Number(spoolDescArr[2]);
        return { jobName, jobId, spoolId };
    }
    /**
     * Helper function that compare-related handlers will use to get the contents of a local file
     * @param filePath Path to the file to compare against
     * @returns Buffer with the contents of the file
     */
    prepareLocalFile(filePath) {
        const localFile = path.isAbsolute(filePath) ? filePath : path.resolve(filePath);
        let lfContentBuf;
        let localFileHandle;
        try {
            try {
                // check if the path given is of a file or not
                localFileHandle = fs.openSync(localFile, 'r');
                if (!fs.fstatSync(localFileHandle).isFile()) {
                    throw new imperative_1.ImperativeError({
                        msg: 'Path is not of a file. Please check the path and try again'
                    });
                }
            }
            catch (error) {
                if (error instanceof imperative_1.ImperativeError)
                    throw error;
                throw new imperative_1.ImperativeError({
                    msg: 'Path not found. Please check the path and try again'
                });
            }
            // reading local file as buffer
            lfContentBuf = fs.readFileSync(localFileHandle);
        }
        finally {
            if (localFileHandle != null)
                fs.closeSync(localFileHandle);
        }
        return lfContentBuf;
    }
    /**
     * This method will prepare the strings for comparison ready
     * @param {string | Buffer } content - Content string or buffer of file 1
     * @param {string | Buffer } content2 - - Content string or buffer of file 2
     * @returns
     * @public
     * @memberof CompareBaseHelper
     */
    prepareContent(content) {
        let contentString = content.toString();
        if (this.seqnum === false) {
            const seqnumlen = 8;
            contentString = content.toString().split("\n").map((line) => line.slice(0, -seqnumlen)).join("\n");
        }
        return contentString;
    }
    /**
     * To get the difference string in terminal or in browser
     * @param {string} string1 - string of file 1 content
     * @param {string} string2 - string of file 2 content
     * @param {IDiffOptions} options
     * @returns {IZosFilesResponse}
     * @public
     * @memberof CompareBaseHelper
     */
    getResponse(string1, string2, options) {
        return __awaiter(this, void 0, void 0, function* () {
            //  CHECKING IF THE BROWSER VIEW IS TRUE, OPEN UP THE DIFFS IN BROWSER
            if (this.browserView) {
                yield imperative_1.DiffUtils.openDiffInbrowser(string1, string2, options);
                return {
                    success: true,
                    commandResponse: "Launching data-sets diffs in browser...",
                    apiResponse: {}
                };
            }
            const jsonDiff = yield imperative_1.DiffUtils.getDiffString(string1, string2, {
                outputFormat: 'terminal',
                contextLinesArg: this.contextLines,
                name1: options === null || options === void 0 ? void 0 : options.name1,
                name2: options === null || options === void 0 ? void 0 : options.name2
            });
            return {
                success: true,
                commandResponse: jsonDiff,
                apiResponse: {}
            };
        });
    }
}
exports.CompareBaseHelper = CompareBaseHelper;
//# sourceMappingURL=CompareBaseHelper.js.map