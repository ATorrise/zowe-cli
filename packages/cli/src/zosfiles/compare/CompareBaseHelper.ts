/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { IHandlerParameters, DiffUtils } from "@zowe/imperative";
import {  IZosFilesResponse } from "@zowe/zos-files-for-zowe-sdk";
import { ICompareFileOptions } from "./doc/ICompareFileOptions";

/**
 * Base helper for compare command to handle common operations through the comparison
 * @export
 */
export default class CompareBaseHelper {


    /**
     * file compare option for file1
     * @public
     * @memberof CompareBaseHelper
     */
    public file1Options: ICompareFileOptions = {} as ICompareFileOptions;

    /**
     * file compare option for file2
     * @public
     * @memberof CompareBaseHelper
     */
    public file2Options: ICompareFileOptions = {} as ICompareFileOptions;

    /**
     * responseTimeout option
     * @public
     * @memberof CompareBaseHelper
     */
    public responseTimeout: number;

    /**
     * seqnum option
     * @public
     * @memberof CompareBaseHelper
     */
    public seqnum: boolean;

    /**
     * contextLines option
     * @private
     * @memberof CompareBaseHelper
     */
    private contextLines: number;

    /**
     * browserView option
     * @private
     * @memberof CompareBaseHelper
     */
    private  browserView: boolean;

    /**
     * Creates an instance of CompareBaseHelper
     * @param {IHandlerParameters} commandParameters Command parameters sent to the handler
     * @memberof CompareBaseHelper
     */
    constructor(commandParameters: IHandlerParameters){
        this.file1Options.binary = commandParameters.arguments.binary;
        this.file2Options.binary = commandParameters.arguments.binary2;
        this.file1Options.encoding = commandParameters.arguments.encoding2;
        this.file2Options.encoding = commandParameters.arguments.encoding2;
        this.file1Options.record = commandParameters.arguments.record2;
        this.file2Options.record = commandParameters.arguments.record2;
        this.file1Options.volumeSerial = commandParameters.arguments.volumeSerial2;
        this.file2Options.volumeSerial = commandParameters.arguments.volumeSerial2;

        if (this.file2Options?.binary == undefined) {
            this.file2Options.binary = this.file1Options.binary;
        }
        if (this.file2Options.encoding == undefined) {
            this.file2Options.encoding = this.file1Options.encoding;
        }
        if (this.file2Options.record == undefined) {
            this.file2Options.record = this.file1Options.record;
        }

        this.seqnum = commandParameters.arguments.seqnum;
        this.browserView = commandParameters.arguments.browserView;
        this.contextLines = commandParameters.arguments.contextLines;
        this.responseTimeout = commandParameters.arguments.responseTimeout;
    }
    /**
     *
     * @param {string | Buffer } content1 - Content string or buffer of file 1
     * @param {string | Buffer } content2 - - Content string or buffer of file 2
     * @returns
     * @public
     * @memberof CompareBaseHelper
     */
    public prepareStrings(content1: string | Buffer, content2: string | Buffer ) {
        let contentString1: string;
        let contentString2: string;

        if(!this.seqnum){
            const seqnumlen = 8;

            const stringArray1 = content1.toString().split("\n");
            for (const i in stringArray1) {
                const sl = stringArray1[i].length;
                const tempString = stringArray1[i].substring(0, sl - seqnumlen);
                contentString1 += tempString + "\n";
            }

            const stringArray2 = content2.toString().split("\n");
            for (const i in stringArray2) {
                const sl = stringArray2[i].length;
                const tempString = stringArray2[i].substring(0, sl - seqnumlen);
                contentString2 += tempString + "\n";
            }
        }
        else {
            contentString1 = content1.toString();
            contentString2 = content2.toString();
        }

        return {
            contentString1, contentString2
        };
    }

    /**
     *
     * @param {string} string1 - string of file 1 comtent
     * @param  {string} string2 - string of file 2 comtent
     * @returns {IZosFilesResponse}
     * @public
     * @memberof CompareBaseHelper
     */
    public async getResponse(string1: string, string2: string): Promise<IZosFilesResponse>{
        //  CHECHKING IIF THE BROWSER VIEW IS TRUE, OPEN UP THE DIFFS IN BROWSER
        if (this.browserView) {

            await DiffUtils.openDiffInbrowser(string1, string2);

            return {
                success: true,
                commandResponse: "Launching data-sets diffs in browser...",
                apiResponse: {}
            };
        }

        let jsonDiff = "";

        jsonDiff = await DiffUtils.getDiffString(string1, string2, {
            outputFormat: 'terminal',
            contextLinesArg: this.contextLines
        });

        return {
            success: true,
            commandResponse: jsonDiff,
            apiResponse: {}
        };
    }
}
