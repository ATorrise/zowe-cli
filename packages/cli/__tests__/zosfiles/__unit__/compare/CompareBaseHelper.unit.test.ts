/*
* This program and the accompanying materials are made available under the terms of the
* Eclipse Public License v2.0 which accompanies this distribution, and is available at
* https://www.eclipse.org/legal/epl-v20.html
*
* SPDX-License-Identifier: EPL-2.0
*
* Copyright Contributors to the Zowe Project.
*
*/

import { CompareBaseHelper } from "../../../../src/zosfiles/compare/CompareBaseHelper";
import {DiffUtils} from "@zowe/imperative";

describe("Comapare Base Helper", () => {

    const fakeCommandParam = {
        arguments: {
            $0: "fake",
            _: ["fake"],
            browserView: false,
        },
    };

    const helper = new CompareBaseHelper(fakeCommandParam as any);
    const string1: string = "test string 1";
    const string2: string = "test string 2";

    helper.prepareContent = jest.fn((string: string | Buffer)=> {
        return "prepared string 1";
    });
    describe("it should prepare the comtent strings for comparison", () => {
        it("should return the prepared strings", async ()=>{
            const returnedStrings = await helper.prepareContent(string1);
            expect(returnedStrings).toContain("prepared");
        });

    });

    describe("it should return the response as per the options passed either in termianl or in browser", () => {
        it("should give response in terminal", ()=>{
            DiffUtils.getDiffString = jest.fn( async () => {
                return "compared string";
            });
            helper.getResponse(string1, string2);
            expect(DiffUtils.getDiffString).toBeCalled();
        });


        it("should initiate the diff opening in browser", ()=>{
            helper.browserView = true;
            jest.spyOn(DiffUtils, "openDiffInbrowser").mockImplementation(jest.fn());
            helper.getResponse(string1, string2);
            expect(DiffUtils.openDiffInbrowser).toBeCalled();
            expect(DiffUtils.openDiffInbrowser).toBeCalledWith(string1, string2);

        });
    });
});
