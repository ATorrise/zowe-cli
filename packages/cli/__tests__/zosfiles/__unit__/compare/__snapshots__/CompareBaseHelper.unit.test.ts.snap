// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comapare Base Helper it should prepare the comtent strings for comparison should return the prepared strings 1`] = `
{
  "contentString1": "undefinedtest 
",
  "contentString2": "undefinedtest 
",
}
`;
